contracts/contracts/strategies/LidoWithdrawalStrategy.sol

```diff
@@ -0,0 +1,307 @@
+    ) external view returns (uint256[] memory hintIds);
+
+    function claimWithdrawals(
+        uint256[] calldata _requestIds,
+        uint256[] calldata _hints
+    ) external;
+
+    function getWithdrawalStatus(uint256[] calldata _requestIds)
+        external
+        view
+        returns (WithdrawalRequestStatus[] memory statuses);
+
+    function getWithdrawalRequests(address _owner)
+        external
+        view
+        returns (uint256[] memory requestsIds);
+
+    function finalize(
+        uint256 _lastRequestIdToBeFinalized,
+        uint256 _maxShareRate
+    ) external payable;
+}
+
+/**
+ * @title Lido Withdrawal Strategy
+ * @notice This strategy withdraws ETH from stETH via the Lido Withdrawal Queue contract
+ * @author Origin Protocol Inc
+ */
+contract LidoWithdrawalStrategy is InitializableAbstractStrategy {
+    IWETH9 private constant weth =
+        IWETH9(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    IERC20 private constant stETH =
+        IERC20(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
+    IStETHWithdrawal private constant withdrawalQueue =
+        IStETHWithdrawal(0x889edC2eDab5f40e902b864aD4d7AdE8E412F9B1);
+    uint256 public constant MaxWithdrawalAmount = 1000 ether;
+    uint256 public outstandingWithdrawals;
+
+    event WithdrawalRequests(uint256[] requestIds, uint256[] amounts);
+    event WithdrawalClaims(uint256[] requestIds, uint256 amount);
+
+    constructor(BaseStrategyConfig memory _stratConfig)
+        InitializableAbstractStrategy(_stratConfig)
+    {
+        require(MaxWithdrawalAmount < type(uint120).max);
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     * @param _rewardTokenAddresses Address of reward token for platform
+     * @param _assets Addresses of initial supported assets
+     * @param _pTokens Platform Token corresponding addresses
+     */
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+        safeApproveAllTokens();
+    }
+
+    /**
+     * @notice deposit() function not used for this strategy. Use depositAll() instead.
+     */
+    function deposit(address, uint256) public override onlyVault nonReentrant {
+        // This method no longer used by the VaultAdmin, and we don't want it
+        // to be used by VaultCore.
+        require(false, "use depositAll() instead");
+    }
+
+    /**
+     * @notice Takes all given stETH and creates Lido withdrawal request
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 stETHStart = stETH.balanceOf(address(this));
+        require(stETHStart > 0, "No stETH to withdraw");
+
+        uint256 numWithdrawals = (stETHStart / MaxWithdrawalAmount) + 1;
```

Comment:
In case `stETHStart mod MaxWithdrawalAmount == 0` the +1 is not necessary.

Reply:
`numWithdrawals` is how many withdrawal requests we are sending to the Lido withdrawal queue.
When the stETH balance is less than the `MaxWithdrawalAmount` (1000 ether), then there is one withdrawal.